### PR TITLE
website: sync recent commits and 1.10.0-beta2

### PR DESCRIPTION
**Pull request by mikemorris**
Syncs a handful of changes that ended up on `stable-website` but not the `release/1.10.x` branch due to some issues with CI and the `type/docs-cherrypick` workflow.

### CODE REVIEW

**Review by BRKalow — COMMENTED** [review]
The diff looks correct, I'll defer to @BrandonRomano as it seems like it is largely from his PR

**Review by BrandonRomano — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/366258.

**Pull request merged**
